Modules/_decimal/_decimal.c

```diff
@@ -1357,6 +1357,7 @@ context_new(PyTypeObject *type, PyObject *args UNUSED, PyObject *kwds UNUSED)
     }
     else {
         self = (PyDecContextObject *)type->tp_alloc(type, 0);
+        assert(PyObject_GC_IsTracked((PyObject *)self));
```

Comment:
Would it not be better to assert this above L1397? You want to assert that the new object is tracked when an object is returned.